### PR TITLE
Adds an option to gracefully handle exceptions.

### PR DESCRIPTION
**Pull request by KapitonovRoman**
Addresses https://github.com/Freshly/stenotype/issues/20 and https://github.com/Freshly/stenotype/issues/11

### CODE REVIEW

File: lib/stenotype/event.rb [hunk 1]
```diff
@@ -20,9 +20,19 @@ class Event
...
+        event.emit!
+        event
+      rescue => error
+        #
+        # @todo This is a temporary solution to enable conditional logger fetching
+        #   needs a fix to use default Spicerack::Configuration functionality
+        #
+        Stenotype::Configuration.logger.error(error)
+
+        raise unless Stenotype.config.graceful_error_handling
```
Comment (mbhnyc) [inline lib/stenotype/event.rb]: dumb ruby question, will this raise call raise the caught error from the block? or will it just raise a generic StandardError?

Comment (mbhnyc) [reply lib/stenotype/event.rb]: if this is getting raise to the world, it should probably be a ST type error

Comment (KapitonovRoman) [reply lib/stenotype/event.rb]: it's going to raise what's caught in the rescue block

Comment (KapitonovRoman) [reply lib/stenotype/event.rb]: @mbhnyc but you are right we better define a single potential rescue point for the client code, i'm going to raise `Stenotype::Error` for this

**Pull request merged**